src/utils/hooks/useTriggerProcessing.tsx

```diff
@@ -3,32 +3,46 @@ import { useGetTriggersQuery } from '../../utils/generated/graphql'
 
 type TriggerType = 'applicationTrigger' | 'reviewAssignmentTrigger' | 'reviewTrigger'
 
-interface TriggerQueryProps {
-  checkTrigger: boolean
-  serialNumber?: string
-  reviewAssignmentId?: number
-  reviewId?: number
-  triggerType?: TriggerType
+interface TriggerApplication {
+  isApplicationLoaded: boolean
+  serialNumber: string
 }
+interface TriggerReview {
+  isReviewLoaded: boolean
+  reviewId: number
+}
+interface TriggerAssignment {
+  reviewAssignmentId: number
+}
+type TriggerQueryProps = { triggerType?: TriggerType } & (
+  | TriggerApplication
+  | TriggerReview
+  | TriggerAssignment
+)
 
-const useTriggerProcessing = ({
-  checkTrigger,
-  serialNumber,
-  reviewAssignmentId,
-  reviewId,
-  triggerType,
-}: TriggerQueryProps) => {
+const useTriggerProcessing = ({ triggerType, ...props }: TriggerQueryProps) => {
   const [isProcessing, setIsProcessing] = useState(true)
   const [triggerError, setTriggerError] = useState<string>()
 
-  // Ensure at least one the primary identifiers is provided
-  if (!serialNumber && !reviewAssignmentId && !reviewId) {
+  // If triggerType not provided, infer it from the supplied ID
+  const inferredTriggerType = inferTriggerType(triggerType, props)
+
+  if (!inferredTriggerType) {
     setIsProcessing(false)
-    setTriggerError('INVALID QUERY')
+    setTriggerError('Type not defined')
```

Comment:
Maybe "Invalid trigger type" might be better, since it could be defined, but not one of the proper ones.

Reply:
Cool. I'll change to that